### PR TITLE
RFC 101: Compiling Recursive Types

### PR DESCRIPTION
**Pull request by ayazhafiz**
[Formatted](https://github.com/roc-lang/rfcs/blob/13608abd9d00857f393d36b0607edc1833706e30/0101-compiling-recursive-types.md)

### CODE REVIEW

File: 0101-compiling-recursive-types.md [hunk 1]
```diff
@@ -0,0 +1,367 @@
...
+   support structural type systems. Other languages, for example Go or C++, do
+   not allow typing `{f: ""}` without wrapping it in a named type constructor;
+   for example `type mytypename struct { f string }` in Go.
+2. 
+   <summary>
+   Compilation to unboxed value representations.
+
+   An unboxed value is compiled to the representation you would likely choose
+   by hand, if you were only optimizing for the runtime cost of a single value.
+   For example, a Roc value `[A {f: U8}]` of type
```
Comment (timotree3) [inline 0101-compiling-recursive-types.md]:
```suggestion
   For example, a Roc value `A {f: 42}` of type
```

File: 0101-compiling-recursive-types.md [hunk 2]
```diff
@@ -0,0 +1,367 @@
...
+  RecursiveType a -> Num *
+```
+
+Please defer discussion of the exact error message's wording to the implementation.
+See below on how to implement this error message.
+
+A suitable re-implementation of this function would be
+
+```roc
+LinkedList a := [Nil, Cons a (RecursiveType a)]
```
Comment (timotree3) [inline 0101-compiling-recursive-types.md]:
```suggestion
LinkedList a := [Nil, Cons a (LinkedList a)]
```

File: 0101-compiling-recursive-types.md [hunk 3]
```diff
@@ -0,0 +1,367 @@
...
+their tag names and recursive ordering of payloads. Records are ordered by
+their field names recursive ordering of their payloads. The lowest-ordered type
+in the recursive chain is assigned the position of recursion.
+
+Such an algorithm would pick the normal form
+
+```
+Rec : [ From2 { from1 : <Rec> }, End ]
+Rec1 : Rec
+Rec2 : Rec
```
Comment (timotree3) [inline 0101-compiling-recursive-types.md]:
```suggestion
Rec2 : { from1: Rec }
```

File: 0101-compiling-recursive-types.md (hunk 1 above)
Comment (ayazhafiz) [reply 0101-compiling-recursive-types.md]: Thanks!

File: 0101-compiling-recursive-types.md (hunk 2 above)
Comment (ayazhafiz) [reply 0101-compiling-recursive-types.md]: Thanks! I should tried to plug all this into the compiler 😅

File: 0101-compiling-recursive-types.md (hunk 3 above)
Comment (ayazhafiz) [reply 0101-compiling-recursive-types.md]: Thanks!

**Pull request merged**